mrmShared/src/mrmSeq.cpp

```diff
@@ -570,6 +590,42 @@ void SoftSequence::disable()
     DEBUG(1, ("Disabled\n") );
 }
 
+void SoftSequence::pause()
+{
+    SCOPED_LOCK(mutex);
+    DEBUG(3, ("Pausing %c\n", is_enabled ? 'Y' : 'N') );
+    if(!is_enabled)
+        {DEBUG(3, ("Skip\n")); return;}
+
+    is_enabled = false;
+
+    if(hw) {
+        interruptLock L;
+        hw->pause();
+    }
+
+    scanIoRequest(changed);
+    DEBUG(1, ("Paused\n") );
+}
+
+void SoftSequence::abort()
+{
+    SCOPED_LOCK(mutex);
+    DEBUG(3, ("Aborting %c\n", is_enabled ? 'Y' : 'N') );
+    if(!is_enabled)
+        {DEBUG(3, ("Skip\n")); return;}
+
+    is_enabled = false;
+
+    if(hw) {
+        interruptLock L;
+        hw->abort();
```

Comment:
Is a `hw->disarm()` need before aborting?  I think this is implied by `is_enabled=false`.

eg. one race condition to consider: What happens when aborting immediate before another trigger?